azconfig/client/src/main/java/com/azure/azconfig/AzConfigClient.java

```diff
@@ -0,0 +1,607 @@
+                       .concatMap(page -> {
+                           String nextPageLink1 = page.nextLink();
+                           if (nextPageLink1 == null) {
+                               return Flux.just(page);
+                           }
+                           return Flux.just(page).concatWith(p -> listLabelsNextAsync(nextPageLink1));
+                       });
+    }
+
+    /**
+     * Gets all Labels.
+     *
+     * @return the Flux&lt;RestPagedResponse&lt;Label&gt;&gt; object if successful.
+     */
+    private Flux<RestPagedResponse<Label>> listLabelsSinglePageAsync(KeyLabelFilter filter) {
+        Mono<RestResponse<Map<String, String>, PageImpl<Label>>> result;
+        if (filter != null) {
+            result = service.listLabels(baseUri.toString(), filter.name(), filter.fields(), filter.acceptDatetime(), filter.range());
+        } else {
+            result = service.listLabels(baseUri.toString(), null, null, null, null);
+        }
+        return result.flatMapMany(p -> Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(), p.headers(), p.statusCode())));
+    }
+
+    private Flux<RestPagedResponse<Label>> listLabelsNextSinglePageAsync(@NonNull String nextPageLink) {
+        if (nextPageLink == null) {
+            throw new IllegalArgumentException("Parameter nextPageLink is required and cannot be null.");
+        }
+        String nextUrl = String.format("%s", nextPageLink);
+        return service.listLabelsNext(baseUri.toString(), nextUrl)
+                       .flatMapMany(p -> Flux.just(new RestPagedResponseImpl<>(p.body().items(), p.body().nextPageLink(), p.request(), p.headers(), p.statusCode())));
+    }
+
+    /**
+     * List all Keys.
+     *
+     * @return keys
+     */
+    public Flux<Key> listKeys(KeyLabelFilter filter) {
+        return listKeys(filter, pageResponseFlux -> pageResponseFlux.map(r -> r.items())
+                                                            .flatMapIterable(i -> i));
+    }
+
+
+    public <T> Flux<T> listKeys(KeyLabelFilter filter, Function<Flux<RestPagedResponse<Key>>, ? extends Flux<T>> receiver) {
+        Flux<RestPagedResponse<Key>> p = listKeysSinglePageAsync(filter)
+                       .concatMap(page -> {
+                           String nextPageLink = page.nextLink();
+                           if (nextPageLink == null) {
+                               return Flux.just(page);
+                           }
+                           return Flux.just(page).concatWith(listKeysNextAsync(nextPageLink));
+                       });
+       return  receiver.apply(p);
+    }
+
+    /**
+     * Gets all Keys.
+     *
+     * @param nextPageLink The nextPageLink from the previous successful call to List operation.
+     * @return the observable to the Page&lt;Key&gt; object.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     */
```

Comment:
It's a bit odd to see JavaDoc on private implementation methods. It is probably a good idea to change from `/**` to `/*`.